### PR TITLE
Flyte Add Enable/Disable Options for Agent Services

### PR DESCRIPTION
**Pull request by Future-Outlier**
# TL;DR
- Added options to enable (default) or disable the Flyte agent.
- Add a flag `--disable-agent`

## Type
- [ ] Bug Fix
- [x] Feature
- [x] Plugin

## Are all requirements met?

- [x] Code completed
- [ ] Smoke tested
- [x] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
@pingsutw and I have introduced two options for users to configure the Flyte agent - it can now be either enabled or disabled.

We believe this enhancement will offer more flexibility and cater to diverse user needs and deployment scenarios.

After a thorough consideration, we have decided to set the "enable agent" option as the default setting.

### Enable agent service (Default)
Enable the agent deployment by updating helm and kustomization.
You can use my image on the docker hub to test.
```
cd flyte/docker/sandbox-bundled/
make build
docker tag flyte-sandbox:latest futureoutlier/flyte-sandbox:flyteagent-v2
docker push futureoutlier/flyte-sandbox:flyteagent-v2
```

```
./bin/flytectl demo start --image futureoutlier/flyte-sandbox:flyteagent-v2
```
### Disable agent service (Use flag)
```
./bin/flytectl demo start --image futureoutlier/flyte-sandbox:flyteagent-v2 --disable-agent
```

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3936
https://github.com/flyteorg/flyte/pull/4041

### CODE REVIEW

File: cmd/config/subcommand/sandbox/sandbox_config.go [hunk 1]
```diff
@@ -19,6 +19,9 @@ type Config struct {
 	// Default value false represents that Flytectl will not use the latest pre-release if it exists.
 	Prerelease bool `json:"pre" pflag:",Optional. Pre release Version of flyte will be used for sandbox."`
 
+	// Agent Service
+	DisableAgent bool `json:"disable-agent" pflag:",Optional. Disable the agent feature."`
```
Comment (pingsutw) [inline cmd/config/subcommand/sandbox/sandbox_config.go:23]: nit
```suggestion
	DisableAgent bool `json:"disable-agent" pflag:",Optional. Disable the agent service."`
```

Comment (Future-Outlier) [reply cmd/config/subcommand/sandbox/sandbox_config.go:23]: Good suggestion, thanks a lot!

**Review by pingsutw — APPROVED** [review]
